src/v/storage/batch_cache.h

```diff
@@ -379,6 +383,7 @@ class batch_cache {
     ss::lowres_clock::time_point _last_reclaim;
     size_t _reclaim_size;
     background_reclaimer _background_reclaimer;
+    resources::deregister_holder _available_mem_deregister;
```

Comment:
Ensures the that reporter gets unregistered when the batch cache is destroyed, so it doesn't dangle.

Reply:
I guess something-something-weak_pointer is another option for this registration idiom.